spring-mvc/src/main/java/com/kodemon/springmvc/controllers/FightController.java

```diff
@@ -0,0 +1,205 @@
+        LOG.debug("Viewing fights of '" + gym.getCity() + "' Gym");
+        return "fight/list";
+    }
+
+    /**
+     * Show list of given user's fights.
+     *
+     * @param username which user's fights do display
+     * @param model data to display
+     * @return JSP page name
+     */
+    @RequestMapping(value = "/listFightsOfUser", method = RequestMethod.GET)
+    public String listFightsOfUser(@RequestParam String username, Model model) {
+
+        Collection<UserDTO> user = userFacade.findUserByUserName(username);
+        if(user.isEmpty()) {
+            model.addAttribute("alert_warning", "No trainer with such username found");
+            LOG.error("Tried to look up fights of '" + username +"' - user doesn't exists.");
+            return "home";
+        }
+        Collection<FightDTO> fights = fightFacade.listFightsOfTrainer(userFacade.findUserByUserName(username).iterator().next());
+        if(fights.isEmpty()) {
+            model.addAttribute("alert_warning", "No fights of this user found");
+            model.addAttribute("trainer", userFacade.findUserByUserName(username).iterator().next());
+            LOG.debug("Tried to look up fights of '" + username +"' - user doesn't have any fight history.");
+            return "/user/detail";
+        }
+        model.addAttribute("fights", fights);
+        LOG.debug("Viewing fights of user '" + username + "'");
+        return "fight/list";
+    }
+
+    @RequestMapping(value = "/grass", method = RequestMethod.GET)
+    public String grass(ServletRequest r, Model model) {
+        HttpServletRequest request = (HttpServletRequest) r;
+        HttpSession session = request.getSession();
+        if ((UserDTO)session.getAttribute("authenticatedUser") == null)
+        {
+            model.addAttribute("alert_warning", "You are not logged in!");
+            LOG.error("User not logged in.");
+            return "home";
+        }
+        Collection<UserDTO> users = userFacade.findUserByUserName(((UserDTO)session.getAttribute("authenticatedUser")).getUserName());
+        UserDTO user = users.iterator().next();
+        PokemonDTO wildPokemon = pokemonFacade.generateWildPokemon(user);
+        session.setAttribute("wildPokemon", wildPokemon);
+        model.addAttribute("wildPokemon", wildPokemon);
+        model.addAttribute("trainersPokemon", user.getPokemons().get(0));
+        LOG.debug("A wild " + wildPokemon.getName().getName() + " level " + wildPokemon.getLevel() + " appeared for " + user.getUserName());
+        return "fight/grass";
+    }
+
+    @RequestMapping(value = "/fightWild", method = RequestMethod.GET)
+    public String fightWild(@RequestParam String mode, ServletRequest r, Model model) {
+        HttpServletRequest request = (HttpServletRequest) r;
+        HttpSession session = request.getSession();
+        WildPokemonFightMode mode_ = (mode.equals("fight")) ? WildPokemonFightMode.TRAIN : WildPokemonFightMode.CATCH;
```

Comment:
catch. I just gave this string 'fight' instead of 'train' because I named the button "FIGHT IT". Doesn't matter really imo, if the button's going to be renamed to something else then of course I will change it.